core/src/main/scala/chisel3/dontTouch.scala

```diff
@@ -46,4 +47,18 @@ object dontTouch {
     data
   }
 
+  /** Mark a module so that its ports will not get optimized away.
```

Comment:
does this already do the right thing w.r.t. property ports and probe ports?

Reply:
As we discussed in dev, we're fine to add this API but we should direct users to `Public` and `Probe` here in the API docs depending on the use case.